pkgs/tools/misc/routino/default.nix

```diff
@@ -9,19 +9,33 @@ stdenv.mkDerivation rec {
     sha256 = "1ccx3s99j8syxc1gqkzsaqkmyf44l7h3adildnc5iq2md7bp8wab";
   };
 
+  patchFlags = [ "-p0" ];
+  patches = stdenv.lib.optionals stdenv.isDarwin [
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/macports/macports-ports/18fd229516a46e7272003acbe555735b2a902db7/gis/routino/files/patch-Makefile_conf.diff";
+      sha256 = "1b7hpa4sizansnwwxq1c031nxwdwh71pg08jl9z9apiab8pjsn53";
+    })
+    (fetchpatch {
+      url = "https://raw.githubusercontent.com/macports/macports-ports/18fd229516a46e7272003acbe555735b2a902db7/gis/routino/files/patch-src_Makefile_dylib_extension.diff";
+      sha256 = "1kigxcfr7977baxdsfvrw6q453cpqlzqakhj7av2agxkcvwyilpv";
+    })
+  ];
+
   nativeBuildInputs = [ perl ];
 
   buildInputs = [ zlib bzip2 ];
 
   outputs = [ "out" "doc" ];
 
+  preConfigure = "LD=$CC";
```

Comment:
1. `stdenv.cc.isClang` is more correct here.
2. Looking into original `routino-3.3.2/Makefile.conf`:
```makefile
# Compilation programs
CC=gcc
CXX=g++
LD=gcc

ifdef CLANG
  ifeq ($(CLANG),1)
    CC=clang
    CXX=clang++
    LD=clang
  endif
endif
```
So, I guess `LD=gcc` would be more align with upstream rather than `LD=ld`. What do you think?

Reply:
What about
```suggestion
  CLANG = stdenv.lib.optionalString stdenv.cc.isClang "1";
```
Then it also builds fine on Linux using clang.

Reply:
Good point, updated.